File: codeforces/betaRound14div2/youngPhotographer/youngPhotographer.py
#!/usr/bin/python
import sys

tmp = raw_input().strip().split(' ')
x = int(tmp[1])
left = 0
right = 1000
ret = 0

for line in sys.stdin:
    tmp = line.strip().split(' ')
    left = max(left, min(int(tmp[0]), int(tmp[1])))
    right = min(right, max(int(tmp[0]), int(tmp[1])))
    if left > right:
        ret = -1
        break

if ret == -1:
    print -1
elif left <= x and x <= right:
    print 0
else:
    print min(abs(x - left), abs(x - right))
    

File: codeforces/betaRound14div2/youngPhotographer/youngPhotographer.cpp
#include <sstream>
#include <string>
#include <vector>
#include <map>
#include <algorithm>
#include <iostream>
#include <cstdio>
#include <cstdlib>
#include <cmath>
#include <utility>
#include <set>
#include <cctype>
#include <queue>
#include <stack>
#include <fstream>
#include <cstring>
#include <numeric>

using namespace std;

int solve(vector <int> a, vector <int> b, int x) {
  int ret = 0;
  int left = 0, right = 1000;
  int i;

  for (i=0; i<a.size(); i++) {
    left = max(left, min(a[i], b[i] ));
    right = min(right, max(a[i], b[i] ));
    if (left > right)
      return -1;
  }

  if (left <= x && x <= right)
    return 0;
  else
    return min(abs(x - left), abs(x - right));

  return ret;
}

int main(void) {
  int n, x;
  int a, b;
  vector <int> va, vb;
  int i;

  cin >> n >> x;

  for (i=0; i<n; i++) {
    cin >> a >> b;
    va.push_back(a);
    vb.push_back(b);
  }

  cout << solve(va, vb, x) << endl;

  return 0;
}
